platform/persistence/platform-persistence-core-impl/src/main/java/org/codice/ddf/persistence/internal/PersistentStoreImpl.java

```diff
@@ -289,26 +264,28 @@ public int delete(String type, String cql) throws PersistenceException {
         return idsToDelete.size();
     }
 
-    private SolrClient getSolrCore(String storeName) {
-        if (coreSolrClients.containsKey(storeName)) {
-            LOGGER.debug("Returning core {} from map of coreSolrClients", storeName);
-            return coreSolrClients.get(storeName);
+    private SolrClient getSolrClient(String storeName) {
+
+        Future<SolrClient> clientFuture = null;
+        SolrClient solrClient = null;
+
+        if (solrFutures.containsKey(storeName)) {
```

Comment:
You should be able to replace this if statement with:

`Future<SolrClient> clientFuture = solrFutures.computeIfAbsent(storeName, storeName -> clientFactory.newClient(storeName));`